unit_testing/cli/test_cli_sct_extract_metric.py

```diff
@@ -1,14 +1,23 @@
-from pytest_console_scripts import script_runner
+import csv
+import numpy as np
+
 import pytest
 import logging
 
+from spinalcordtoolbox.scripts import sct_extract_metric
+
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.script_launch_mode('subprocess')
-def test_sct_extract_metric_backwards_compat(script_runner):
-    ret = script_runner.run('sct_testing', '--function', 'sct_extract_metric')
-    logger.debug(f"{ret.stdout}")
-    logger.debug(f"{ret.stderr}")
-    assert ret.success
-    assert ret.stderr == ''
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_extract_metric_weighted_average_value_against_groundtruth():
+    """Verify that computed WA value is equivalent to known ground truth value."""
+    fname_out = 'quantif_mtr.csv'
+    sct_extract_metric.main(argv=['-i', 'mt/mtr.nii.gz', '-f', 'mt/label/atlas', '-method', 'wa', '-l', '51',
+                                  '-z', '1:2', '-o', fname_out])
+    with open(fname_out, 'r') as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=',')
+        mtr_result = np.float([row['WA()'] for row in reader][0])
+
+    assert mtr_result == pytest.approx(32.6404)
```

Comment:
same thing with pytest.approx bound

Reply:
Sorry about this! Thanks for catching another instance. :sweat_smile:

Reply:
Done in 16261d44.